src/components/MonthlyGood/DropInPaymentWrapperMG.vue

```diff
@@ -0,0 +1,460 @@
+import LoadingOverlay from '@/pages/Lend/LoadingOverlay';
+
+export default {
+	components: {
+		KvButton,
+		KvIcon,
+		LoadingOverlay,
+	},
+	inject: ['apollo'],
+	mixins: [
+		checkoutUtils
+	],
+	props: {
+		amount: {
+			type: Number,
+			default: 0
+		},
+		donateAmount: {
+			type: Number,
+			default: 0
+		},
+		dayOfMonth: {
+			type: Number,
+			default: 0
+		},
+		category: {
+			type: String,
+			default: ''
+		},
+		isOneTime: {
+			type: Boolean,
+			default: false
+		}
+	},
+	data() {
+		return {
+			enableCheckoutButton: false,
+			btDropinInstance: () => {},
+			clientToken: null,
+			updatingPaymentWrapper: false,
+		};
+	},
+	methods: {
+		setUpdatingPaymentWrapper(state) {
+			this.updatingPaymentWrapper = state;
+		},
+		initializePaymentOptions() {
+			this.setUpdatingPaymentWrapper(true);
+			this.apollo.query({
+				query: getClientToken,
+				variables: {
+					amount: numeral(this.amount).format('0.00'),
+					useCustomerId: true
+				}
+			}).then(response => {
+				if (response.errors) {
+					this.setUpdatingPaymentWrapper(false);
+					console.error(response.errors);
+					const errorCode = _get(response, 'errors[0].code');
+					const errorMessage = _get(response, 'errors[0].message');
+
+					Sentry.withScope(scope => {
+						scope.setTag('bt_stage_dropin', 'btGetClientTokenError');
+						scope.setTag('bt_get_client_token_error', errorMessage);
+						Sentry.captureException(errorCode);
+					});
+				} else {
+					this.clientToken = _get(response, 'data.shop.getClientToken');
+					this.initializeDropIn();
+				}
+			});
+		},
+		initializeDropIn() {
+			Dropin.create({
+				authorization: this.clientToken,
+				container: '#dropin-container',
+				dataCollector: {
+					kount: true // Required if Kount fraud data collection is enabled
+				},
+				card: {
+					vault: {
+						allowVaultCardOverride: true,
+					},
+				},
+				paypal: {
+					flow: 'vault',
+					amount: numeral(this.amount).format('0.00'),
+					currency: 'USD',
+					buttonStyle: {
+						color: 'gold',
+						shape: 'rect',
+						size: 'responsive',
+					}
+				},
+			}).then(btCreateInstance => {
+				this.btDropinInstance = btCreateInstance;
+				this.initializeDropInActions();
+				this.setUpdatingPaymentWrapper(false);
+			}).catch(btCreateError => {
+				console.error(btCreateError);
+				Sentry.withScope(scope => {
+					scope.setTag('bt_stage_dropin', 'btCreateError');
+					scope.setTag('bt_client_create_error', btCreateError.message);
+					Sentry.captureException(btCreateError.code);
+				});
+			});
+		},
+		submitDropInPayment() {
+			this.$kvTrackEvent('Registration', 'click-confirm-monthly-good', 'register-monthly-good');
+
+			// request payment method
+			this.btDropinInstance.requestPaymentMethod()
+				.then(btSubmitResponse => {
+					const transactionNonce = _get(btSubmitResponse, 'nonce');
+					const deviceData = _get(btSubmitResponse, 'deviceData');
+					const paymentType = _get(btSubmitResponse, 'type');
```

Comment:
not sure we need to pass deviceData any longer but we should push through the paymentType and tie it into the analytics event.